### PR TITLE
KV Secret LIST delete and permissions

### PR DESCRIPTION
**Pull request by Monkeychip**
I originally had these all in another PR, but moved them out to keep the file changes at a reasonable number.

This PR:
- [x] adds Metadata DELETE action on the list popup menu.
- [x] adds Permissions on list popup menu items.
- [x] fixes some Breadcrumbs.

https://github.com/hashicorp/vault/assets/6618863/54f17a3d-bc13-4de4-9715-64dc32871de7

### CODE REVIEW

File: ui/lib/kubernetes/addon/components/page/roles.js
Comment (Monkeychip) [inline ui/lib/kubernetes/addon/components/page/roles.js:1]: I couldn't help myself. These needed periods.

File: ui/app/lib/kv-breadcrumbs.js [hunk 1]
```diff
@@ -13,13 +13,13 @@ function pathIsFromDirectory(path) {
   return path ? !!path.match(/\//) : false;
 }
 
-function breadcrumbsForDirectory(path) {
+function breadcrumbsForDirectory(path, noLinkToLastItem) {
```
Comment (kiannaquach) [inline ui/app/lib/kv-breadcrumbs.js:16]: Could we default `noLinkToLashItem = false` if it's not passed in?

Comment (kiannaquach) [inline ui/app/lib/kv-breadcrumbs.js:16]:
```suggestion
function breadcrumbsForDirectory(path, noLinkToLastItem = false) {
```

Comment (kiannaquach) [reply ui/app/lib/kv-breadcrumbs.js:16]: It might be easier to understand that by default if `noLinkToLastItem` is not passed in, it defaults to false.

**Review by kiannaquach — APPROVED** [review]

**Pull request merged**